strawberry/cli/commands/upgrade/_run_codemod.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations
+
+import contextlib
+import os
+from multiprocessing import Pool, cpu_count
+from typing import TYPE_CHECKING, Any, Dict, Generator, Sequence, Union
+
+from libcst.codemod._cli import ExecutionConfig, ExecutionResult, _execute_transform
+from libcst.codemod._dummy_pool import DummyPool
+from rich.progress import Progress
+
+from ._fake_progress import FakeProgress
+
+if TYPE_CHECKING:
+    from libcst.codemod import Codemod
+
+ProgressType = Union[type[Progress], type[FakeProgress]]
+PoolType = Union[type[Pool], type[DummyPool]]  # type: ignore
+
+
+def _execute_transform_wrap(
+    job: Dict[str, Any],
+) -> ExecutionResult:
+    # TODO: maybe capture warnings?
```

Comment:
warnings happen when libcst failed to determine module name for the file, I was getting this a few times, not sure if it is important or not